### PR TITLE
Sheffield | 26-ITP-Jan | Mahmoud Shaabo | Sprint 3 | ToDo List App

### PR DESCRIPTION
**Pull request by mahmoudshaabo1984**
Hello Reviewers / CodeYourFuture Team,

I have completed the ToDo List App project for Sprint 3. I am excited to share that I have successfully implemented the mandatory requirement for the mass delete feature.

🎯 **Acceptance Criteria Met:**

- [x] **UI Update:** Added a "Delete completed tasks" button to `index.html` with the correct ID.
- [x] **Logic Implementation:** Created the `deleteCompleted(todoList)` function in `todos.mjs` to remove all completed tasks.
- [x] **Testing:** Wrote passing Jest tests in `todos.test.mjs` to verify the logic works correctly (including edge cases).
- [x] **Integration:** Connected the new button to the `deleteCompletedTodos` function in `script.mjs` and successfully updated the DOM.

💻 **Technical Implementation Details:**

- Utilized a reverse `for` loop in the `deleteCompleted` function. This approach ensures that modifying the array length during iteration does not cause index shifting or skip any items, which is a common issue when using `splice()` in a standard forward loop.
- Ensured all modular imports/exports work correctly as ES Modules.
- All Jest tests (`npm test`) are passing successfully.

Thank you for your time and feedback!

Best regards,
Mahmoud Shaabo

### CODE REVIEW

**Comment by mahmoudshaabo1984** [comment]
Hi @cifarquhar,

As I submit my Pull Request for the ToDo List App, I want to take a moment to thank you for all your efforts and the valuable knowledge you share with us in this course. 

For this task, I have successfully implemented the mass delete feature for completed ToDos, making sure the business logic in `todos.mjs` is correctly separated from the UI in `script.mjs`. All Jest tests are also passing successfully.

Thank you again for your time and continuous support!

Best regards,
Mahmoud Shaabo

File: Sprint-3/todo-list/todos.mjs [hunk 1]
```diff
@@ -26,4 +26,12 @@ export function toggleCompletedOnTask(todos, taskIndex) {
...
\ No newline at end of file
+}
+// Remove all tasks where completed is true
+export function deleteCompleted(todos) {
+  for (let i = todos.length - 1; i >= 0; i--) {
+    if (todos[i].completed === true) {
+      todos.splice(i, 1);
+    }
+  }
+}
```
Comment (Luro91) [inline Sprint-3/todo-list/todos.mjs:37]: This works correctly. You can also research build in array functions. There is a function that allows you filter elements in an array based on a condition.

**Review by Luro91 — COMMENTED** [review]
Well done. The implementation works as expected

**Comment by illicitonion** [comment]
Closing PR because the January ITP run has finished. Feel free to re-open if you're still working on it.
